app/src/main/java/emu/skyline/AppUpdater.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * SPDX-License-Identifier: MPL-2.0
+ * Copyright © 2023 Skyline Team and Contributors (https://github.com/skyline-emu/)
+ */
+
+package emu.skyline
+
+import android.app.DownloadManager
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Handler
+import android.os.Looper
+import android.text.Html
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat.startActivity
+import org.json.JSONArray
+import org.json.JSONTokener
+import java.io.File
+import java.io.IOException
+import java.net.URL
+import java.util.concurrent.Executors
+
+class AppUpdater : BroadcastReceiver() {
+    private var downloadID = 0L
+    private val baseUrl = "https://skyline-builds.alula.gay"
+    private val branch = "ftx1"
+
+    fun downloadApk(applicationContext : Context) {
+        val myExecutor = Executors.newSingleThreadExecutor()
```

Comment:
You should definitely not use an `Executor` here, probably worth looking into coroutines as they're much easier to work with (once you understand them), and they make code so much cleaner and less error prone.